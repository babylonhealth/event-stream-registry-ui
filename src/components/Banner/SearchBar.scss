@import "../EventStreamRegistry/_variables.scss";

input {
  outline: 0;
  font-family: "Courier New", sans-serif;
  font-size: 12px;
  border-color: $lightGrey;
  border-width: 0 0 1px;
  padding-bottom: 4px;
  padding-left: $lpad;
  height: 12px;
  width: 100%
}
