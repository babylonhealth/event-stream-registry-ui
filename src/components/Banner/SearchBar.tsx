import * as React from "react";

import "./SearchBar.scss"


type SearchBarProps = {
    text: string,
    onTextChange: (x: React.ChangeEvent<HTMLInputElement>) => void
}

/**
 * Renders a search bar that allows users to filter nodes and change the Event Stream Registry view
 * @param {string} text - the input text
 * @param {(x) => void} onTextChange - the action triggered on a change of the input
 * @constructor
 */
const SearchBar = ({text, onTextChange}: SearchBarProps) =>
        <input type="text"
               value={text}
               placeholder="SEARCH... (or type help)"
               spellCheck={false}
               autoFocus
               aria-label={"search-bar"}
               onChange={onTextChange}/>;

export default SearchBar;
