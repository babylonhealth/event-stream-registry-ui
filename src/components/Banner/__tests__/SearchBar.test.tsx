import React from 'react'
import {cleanup, fireEvent, render, screen} from '@testing-library/react'
import SearchBar from "../SearchBar";

afterEach(cleanup);

it('should take a snapshot', () => {
    const {asFragment} = render(<SearchBar text={""} onTextChange={jest.fn()}/>);

    // @ts-ignore
    expect(asFragment(<SearchBar text={""} onTextChange={jest.fn()}/>)).toMatchSnapshot()
});

it('should set button clicked based on view', () => {
    const onTextChange = jest.fn();
    render(<SearchBar text={""} onTextChange={onTextChange}/>);
    const input = screen.getByLabelText("search-bar")
    expect(input.placeholder).toBe("SEARCH... (or type help)");
    fireEvent.click(input, {target: {value: "blah"}});
    expect(input.value).toBe("blah");
});
