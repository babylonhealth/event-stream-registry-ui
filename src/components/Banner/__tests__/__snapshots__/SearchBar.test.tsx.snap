// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <input
    aria-label="search-bar"
    placeholder="SEARCH... (or type help)"
    spellcheck="false"
    type="text"
    value=""
  />
</DocumentFragment>
`;
